collect_app/src/main/java/org/odk/collect/android/utilities/ProjectResetter.kt

```diff
@@ -62,10 +64,18 @@ class ProjectResetter(
     }
 
     private fun resetInstances() {
-        instancesRepositoryProvider.get().deleteAll()
+        changeLockProvider.getInstanceLock(projectId).withLock { acquiredLock: Boolean ->
+            if (acquiredLock) {
+                instancesRepositoryProvider.get().deleteAll()
```

Comment:
Is there a reason not to use `InstancesDataService` here instead?

Reply:
We can do that. Done.